src/Core/Services/Implementations/LaunchDarklyFeatureService.cs

```diff
@@ -8,12 +9,38 @@ public class LaunchDarklyFeatureService : IFeatureService, IDisposable
     private readonly LdClient _client;
 
     public LaunchDarklyFeatureService(
-        GlobalSettings globalSettings)
+        IGlobalSettings globalSettings)
     {
-        var ldConfig = Configuration.Builder(globalSettings.LaunchDarkly?.SdkKey)
-                .Offline(globalSettings.SelfHosted)
-                .Build();
-        _client = new LdClient(ldConfig);
+        var ldConfig = Configuration.Builder(globalSettings.LaunchDarkly?.SdkKey);
+
+        if (string.IsNullOrEmpty(globalSettings.LaunchDarkly?.SdkKey))
+        {
+            // support a file to load flag values
+            const string flagOverridePath = "flags.json";
+            if (File.Exists(flagOverridePath))
+            {
+                ldConfig.DataSource(
+                    FileData.DataSource()
+                        .FilePaths(flagOverridePath)
+                        .AutoUpdate(true)
+                );
+
+                // do not provide analytics events
+                ldConfig.Events(Components.NoEvents);
+            }
+            else
+            {
+                // when a file-based fallback isn't available, work offline
+                ldConfig.Offline(true);
+            }
+        }
+        else if (globalSettings.SelfHosted)
+        {
+            // when self-hosted, work offline
```

Comment:
This does not guarantee offline when self hosted.

Self host is capable of overriding any setting.
```suggestion
        
        if (globalSettings.SelfHosted)
        {
            // when self-hosted, work offline
```

Reply:
My first round in this PR had it your suggested way, but I revised it afterward because I thought it'd be best to support the ability for self-hosted installs to use the file fallback. With the file fallback configured and offline mode set to true, the file will actually be ignored which is a shame, but the lack of event-sending effectively makes it offline.

Put another way, I don't want to explicitly set the offline property to true when self-hosted.

Reply:
In that case, why reference self host at all?

This logic allows for cloud to be put in a bad state should sdk and file be null, just like self host. It seems like the bounds of the SDK are such that if sdk key and file fallback is null, you _must_ work in offline mode
```suggestion
        else
        {
            // fallback to offline
```